modules/managers/galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.cemt.manager/src/main/java/dev/galasa/cicsts/cemt/internal/CemtImpl.java

```diff
@@ -366,5 +449,164 @@ public boolean performSystemProperty(@NotNull ICicsTerminal terminal, @NotNull S
 
     }
 
+    @Override
+    public void waitForDisabledResource(ICicsTerminal terminal, String resourceType, String resourceName,
+        int resourceTimeout)
+            throws CemtException {
+
+        try {
+            long timeoutTimeInMilliseconds = Calendar.getInstance()
+                .getTimeInMillis() + resourceTimeout;
+            while (Calendar.getInstance().getTimeInMillis() < timeoutTimeInMilliseconds) {
+                terminal.resetAndClear();
+                terminal.type("CEMT INQUIRE " + resourceType + "(" + resourceName + ")").enter().waitForKeyboard();
+                if (terminal.retrieveScreen().contains(" Dis ")) {
+                    return;
+                } 
+                if (Calendar.getInstance().getTimeInMillis() >= timeoutTimeInMilliseconds) {
+                    break;
+                }
+                Thread.sleep(1000);
+            }
+        } catch (CicstsManagerException e) {
+            throw new CemtException(
+                "Problem with starting the CEMT transaction", e);
+
+        }
+        catch (Exception e) {
+            throw new CemtException("Unable to prepare for the CEMT inquire resource", e);
+        }
+        
+        throw new CemtException("Timeout of " + resourceTimeout
+            + "ms exceeded while waiting for " + resourceType + "("
+            + resourceName + ") to be disabled");
+    }
+
+    @Override
+    public void waitForEnabledResource(ICicsTerminal terminal, String resourceType, String resourceName)
+        throws CemtException
+    {
+        waitForEnabledResource(terminal, resourceType, resourceName, this.defaultResourceTimeout);
+    }
+
+    @Override
+    public void waitForEnabledResource(ICicsTerminal terminal, String resourceType, String resourceName,
+        int resourceTimeout) throws CemtException
+    {
+        // Calculate when we should timeout
+        long timeoutTimeInMilliseconds = Calendar.getInstance().getTimeInMillis() + resourceTimeout;
+
+        // Keep going until we reach timeout time
+        while (Calendar.getInstance().getTimeInMillis() < timeoutTimeInMilliseconds) {
+          try {
+            // Clear the terminal and issue the inquire command
+            terminal.resetAndClear();
+            terminal.type("CEMT INQUIRE " + resourceType + "(" + resourceName + ")").enter().waitForKeyboard();
+            
+            if (terminal.retrieveScreen().contains(" Ena "))
+            {
+                return;
+            }
+
+            // Double check the timeout time, just in case the the send text
+            // took ages
+            if (Calendar.getInstance().getTimeInMillis() >= timeoutTimeInMilliseconds)
```

Comment:
Same comment here:

The `Calendar.getInstance().getTimeInMillis()` is slightly slower than doing a `System.currentTimeMillis()` or `Instant.now()` due to the need to get a `Calendar` object, so maybe we could use one of those instead?